ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouter.java

```diff
@@ -469,11 +475,22 @@ private void onResponse(List<ResponseInfo> responseInfoList) {
      */
     @Override
     public void run() {
+      long startTime;
+      long lastTime;
       try {
         while (isOpen.get()) {
+          startTime = time.milliseconds();
+          lastTime = startTime;
           List<RequestInfo> requestInfoList = pollForRequests();
+          routerMetrics.operationControllerPollforRequestTimeMs.update(time.milliseconds() - lastTime);
+          lastTime = time.milliseconds();
           List<ResponseInfo> responseInfoList = networkClient.sendAndPoll(requestInfoList);
+          routerMetrics.numOfRequestInfosPerPoll.update(requestInfoList.size());
+          routerMetrics.operationControllerSendAndPollTimeMs.update(time.milliseconds() - lastTime);
```

Comment:
I did see this is tracked by `NetworkClient`. However, it could be possible that multiple entities uses `NetworkClient`, and by only looking into the network-client-level metrics will not report the time that is only relevant to operation controller.